vnext/ReactUWP/Utils/RectUtils.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include <Utils/RectUtils.h>
+
+namespace winrt {
+using namespace Windows::Foundation;
+} // namespace winrt
+
+namespace react {
+namespace uwp {
+
+// Represents positive infinity.
+const float PositiveInfinity = std::numeric_limits<float>::infinity();
+
+// Represents negative infinity.
+const float NegativeInfinity = -std::numeric_limits<float>::infinity();
+
+bool IsPositiveInfinity(float value) {
+  return value == PositiveInfinity;
+}
+
+bool IsNegativeInfinity(float value) {
+  return value == NegativeInfinity;
+}
+
+// This is a read-only alias for X.  If this is the empty rectangle,
+// the value will be positive infinity.
+float GetLeft(_In_ winrt::Rect rect) {
+  return rect.X;
+}
+
+// This is a read-only alias for Y.  If this is the empty rectangle,
+// the value will be positive infinity.
+float GetTop(_In_ winrt::Rect rect) {
+  return rect.Y;
+}
+
+// This is a read-only alias for X + Width.  If this is the empty
+// rectangle, the value will be negative infinity.
+float GetRight(_In_ winrt::Rect rect) {
+  return (IsEmptyRect(rect) ? NegativeInfinity : rect.X + rect.Width);
+}
+
+// This is a read-only alias for Y + Height.  If this is the empty
+// rectangle, the value will be negative infinity.
+float GetBottom(_In_ winrt::Rect rect) {
+  return (IsEmptyRect(rect) ? NegativeInfinity : rect.Y + rect.Height);
+}
+
+REACTWINDOWS_API_(bool) IsEmptyRect(winrt::Rect rect) {
+  return rect.Width < 0;
+}
+
+REACTWINDOWS_API_(winrt::Rect) CreateEmptyRect() {
+  winrt::Rect result;
+  result.X = std::numeric_limits<float>::infinity();
+  result.Y = std::numeric_limits<float>::infinity();
+  result.Width = -std::numeric_limits<float>::infinity();
+  result.Height = -std::numeric_limits<float>::infinity();
+  return result;
+}
+
+// Update this rectangle to be the union of this and rect.
+REACTWINDOWS_API_(winrt::Rect) UnionRect(winrt::Rect rect1, _In_ winrt::Rect rect2) {
+  winrt::Rect result = rect1;
+  if (IsEmptyRect(rect1)) {
```

Comment:
can you just call the user32 UnionRect?
UnionRect(ToRECT(rect1), ToRECT(rect2))